### ISSUE TITLE
Improve mobile navigation for screen reader users by placing `navOpen` button inside `nav` element

### ISSUE THREAD

**Issue by FennaKort** [FennaKort/php-and-accessible-menu-learning#9]
The mobile site navigation markup could be improved by moving the `navOpen` button inside the `nav` element so that screen reader users can jump to the `Navigation` landmark when the navigation is closed on mobile. Additional markup and styling changes are likely necessary to accommodate this. 

> # On accessible off-screen navigation
> https://www.makethingsaccessible.com/guides/accessible-basic-disclosure-widgets/#lets-build-a-mobile-navigation
> 
> https://adrianroselli.com/2019/06/link-disclosure-widget-navigation.html
> 
> ## [Inclusive Components: Navigation menu buttons](https://inclusive-components.design/menus-menu-buttons/#navigationmenubuttons) (hamburger icons for offscreen navigation)
> Ok interesting, this page says the `navOpen` button should be INSIDE the `nav` element so that screen reader users can jump to the `nav` landmark. 
> 
> And yes, the way my HTML is currently written doesn't allow jumping to the `nav` landmark because the `navOpen` button is OUTSIDE the `nav` element, and the `nav` element is being hidden because the `closeMobileMenu` function sets `[inert=""]` on nav#topnavMenu. Should `inert` be set on `ul` instead when `topnavMenu` is closed?
> <img width="701" height="484" alt="Accessible document does not allow for jumping to nav landmark because nav element is hidden/inert and 'Open Navigation' button is outside of nav element." src="https://gith ub.com/user-attachments/assets/facc33f9-3977-4eca-be12-77867543e4eb" />
> 
> Article recommends setting `hidden` on `nav>ul` and toggling state on clicking `nav>button`. Should I use the same button for opening/closing the nav then?
> 
> idly wondering if i should rethink my id/class naming schema to reflect that the site nav is not technically a menu lol
> 
> also, apparently `aria-controls` is only supported in JAWS and not other screen readers, so the expanded nav list's first link should be immediately after the `navOpen` button in the markup so that it receives focus next in order after the button. [See related issue 8](https://github.com/FennaKort/php-and-accessible-menu-learning/issues/8) 

 _Originally posted by @FennaKort in [#7](https://github.com/FennaKort/php-and-accessible-menu-learning/issues/7#issuecomment-3224481179)_

**Comment by FennaKort** [comment]
# [Adrian Roselli: Disclosure Widget Navigation](https://adrianroselli.com/2019/06/link-disclosure-widget-navigation.html)
need further notes and reading but based on a quick skim, I think I can actually simplify the mobile navigation further for an improved experience. Especially interested in using one singular button, and avoiding show/hide or open/close aria labels in favour of the functionality implicitly provided by `aria-expanded` states.

**Comment by FennaKort** [comment]
Would also be interested in improving conformity between styling and use states by reworking some css to function off of the `aria-expanded` states.

**Comment by FennaKort** [comment]
# [Make Things Accessible: Accessible Basic Disclosure Widgets](https://www.makethingsaccessible.com/guides/accessible-basic-disclosure-widgets/#lets-build-a-mobile-navigation)
1. start with basic `nav>ul>li` pattern
2. add button with javascript 
  - could write it to add button when mobile breakpoint is crossed. I'm not sure why the article adds the button with js regardless of whether or not the device is mobile; the article's approach is to always add the button with js, and to use css to set `display: none` on the button at desktop viewport sizes. In this case, why fuss with using js to add the button, rather than just having it in the html?
  - if we give the button text content such as `<button>Menu</button>`, then screenreaders can encounter `nav>button[aria-expanded=false]{Menu}` as something like "navigation, Menu button, collapsed"
    - so therefore we could also give the button an `aria-label` to accessibly name it while providing non-textual icons as visuals?
3. use js to add event listener to set aria-expanded values on click
4. use css to style using aria-expanded values as selectors
5. use js to make the panel dismissible on `esc` press and return focus to opening trigger
6. use js to set `aria-current` to indicate current page; use `aria-current` as your css selector instead of a `class="current-page"` class in the html
